src/_tkagg.cpp

```diff
@@ -293,85 +286,72 @@ void load_tkinter_funcs(void)
  * dynamic library (module).
  */
 
-void load_tkinter_funcs(void)
+static void
+clean_library(void *lib)
+{
+    // We don't need to keep a reference open as the main program & tkinter have been
+    // imported. This cleans up the shared_ptr when it goes out of scope.
+    if (lib && dlclose(lib)) {
+        throw std::runtime_error(dlerror());
```

Comment:
IIUC, one should not throw in a destructor, but this is the deleter function for a `shared_ptr`; I'm not sure if that also counts or not. It probably does and I'll have to change this.

Reply:
I agree, from https://en.cppreference.com/w/cpp/memory/shared_ptr/shared_ptr:
```
Uses the specified deleter d as the deleter. The expression d(ptr) must be well formed, have well-defined behavior and not throw any exceptions. The construction of d and of the stored deleter copied from it must not throw exceptions. 
```
The other elegant C++ solutions to this that I thought of end up with the same problem as they reduce to trying to throw an exception in a destructor. Maybe it might be easiest to `dlclose` explicitly at the end of the function below which won't be as elegant but is no worse than it used to be.

Reply:
I re-arranged things a bit; we don't actually need the libraries open over the entire extent of the function, so that really reduces the need for the (not working) destructor approach entirely.